src/client/auth/scram.rs

```diff
@@ -57,19 +57,32 @@ struct CacheEntry {
 }
 
 /// The versions of SCRAM supported by the driver (classified according to hash function used).
-#[derive(Hash, Eq, PartialEq, Clone)]
+#[derive(Debug, Hash, Eq, PartialEq, Clone)]
 pub(crate) enum ScramVersion {
     Sha1,
     Sha256,
 }
 
+/// Contains the authentication info derived from a Credential.
+pub(crate) struct ClientAuthInfo<'a> {
+    pub(crate) username: &'a str,
```

Comment:
The auth info is all derived from the Credential, which is passed in by reference to the methods below. To take ownership, we'd have to copy them, and while it's not a huge deal since they should be relatively small, the usage of lifetimes here doesn't complicate things too much, so I figured I would just do the efficient thing.